plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergTableOperations.java

```diff
@@ -160,12 +160,13 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
         commitTableUpdate(
                 getTable(database, tableName, false),
                 metadata,
-                (_, newMetadataLocation) ->
+                (table, newMetadataLocation) ->
                         getTableInput(
                                 typeManager,
                                 tableName,
                                 owner,
                                 metadata,
+                                table.getStorageDescriptor().getLocation(),
```

Comment:
Note: `table.getStorageDescriptor().getLocation()` might be different from `metadata.location()`.

Reply:
When does this happen?

Reply:
It happens if users directly modify the location in Glue.